consensus/istanbul/backend.go

```diff
@@ -48,8 +48,10 @@ type Backend interface {
 	// BroadcastConsensusMsg sends a message to all validators (include self)
 	BroadcastConsensusMsg(validators []common.Address, payload []byte) error
 
-	// Gossip sends a message to all validators (exclude self)
-	Gossip(validators []common.Address, payload []byte, ethMsgCode uint64, ignoreCache bool) error
```

Comment:
Great question.  I had a meeting with @timmoreton to chat about this.  The main purpose of removal of this is that only gossiped messages will now be cached.

All direct messages will not be since they are not meant to be duplicated (which gossiped messages could be).  E.g. Tim made changes so that istanbul will periodically resend the same round change message within the same round.  Caching those types of messages will prevent those intended resends.

Reply:
Let me know if you have any concerns with that change.